docs/signer.rst

```diff
@@ -3,12 +3,71 @@ Signer API
 
 .. currentmodule:: securesystemslib.signer
 
+At its core the signer API defines abstract interfaces to create and verify
+cryptographic signatures:
+
+* ``Signer.sign``
+* ``Key.verify_signature``
+
+The interfaces can be implemented using arbitrary asymmetric signing technology.
+The ``Key`` class is also a container class for public key data. The ``Signer``
+class, on the other hand, treats the private key as implementation detail. This
+means that one specific signer may indeed contain the private key, but another
+calls a remote cloud KMS, or a local hardware token for signing.
+
+In addition to sign and verify interface methods, the signer API provides
+generic *load* methods:
+
+* ``Signer.from_priv_key_uri`` -  Loads any specific signer from a URI. The
+  specific signer implementation itself is responsible for the URI format and
+  resolution. To become discoverable, signers and their URI schemes are
+  registered in the ``SIGNER_FOR_URI_SCHEME`` lookup table.
+
+* ``Key.from_dict`` - Loads any specific key from a serialized format. The
+  specific key implementation is responsible public key format and
+  deserialization. To become discoverable, key type and signing scheme --
+  required fields in any public key -- are registered in the
+  ``KEY_FOR_TYPE_AND_SCHEME`` lookup table.
+
+An application can use these same methods to uniformly load any signer or public
+key, regardless of the specific implementation. Many signers and keys are
+already included in the signer API. And an application can also create and
+register its own.
+
+
+Usage
+-----
+The signer API is streamlined for the following series of user events,
+which may happen on different systems and at different points in time:
```

Comment:
I wonder if it's useful to make the connection to TUF explicitly: that the importance of separating these steps comes from the core use case (could maybe even explain the back-and-and-forth of a yubikey holder and a TUF repository when becoming a signer).

Maybe this should be a separate section that explains the design decisions?

Reply:
I was going back and forth about mentioning TUF or not, and ended up without because I didn't want to explain what TUF is, nor assume knowledge about it.

Let's reconsider in a second iteration.